apps/prairielearn/src/pages/studentAssessment/studentAssessment.html.ts

```diff
@@ -0,0 +1,246 @@
+                      </strong>
+                    </p>
+                  `
+                : ''}
+              ${assessment.group_work
+                ? StudentGroupControls({ resLocals })
+                : StartAssessmentForm({ assessment, user, __csrf_token, startAllowed: true })}
+            </div>
+            <!-- card-body -->
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+function StartAssessmentForm({
+  assessment,
+  user,
+  __csrf_token,
+  startAllowed,
+}: {
+  assessment: Assessment;
+  user: User;
+  __csrf_token: string;
+  startAllowed: boolean;
+}) {
+  return html`
+    ${startAllowed && assessment.type === 'Exam' && assessment.require_honor_code
+      ? HonorPledge({ user, groupWork: !!assessment.group_work })
+      : ''}
+    <form
+      id="confirm-form"
+      name="confirm-form"
+      method="POST"
+      class="mt-4 d-flex justify-content-center"
+    >
+      <input type="hidden" name="__action" value="new_instance" />
+      <input type="hidden" name="__csrf_token" value="${__csrf_token}" />
+      <button
+        id="start-assessment"
+        type="submit"
+        class="btn btn-primary"
+        ${(assessment.type === 'Exam' && assessment.require_honor_code) || !startAllowed
+          ? 'disabled'
+          : ''}
+      >
+        Start assessment
+      </button>
+    </form>
+  `;
+}
+
+function HonorPledge({ user, groupWork }: { user: User; groupWork: boolean }) {
+  return html`
+    <div class="card card-secondary mb-4 test-class-honor-code">
+      <ul class="list-group list-group-flush">
+        <li class="list-group-item py-2">
+          I certify that I am ${user.name} and ${groupWork ? 'our group is' : 'I am'} allowed to
+          take this assessment.
+        </li>
+        <li class="list-group-item py-2">
+          ${groupWork ? 'We' : 'I'} pledge on my honor that ${groupWork ? 'we' : 'I'} will not give
+          or receive any unauthorized assistance on this assessment and that all work will be
+          ${groupWork ? 'our' : 'my'} own.
+        </li>
+      </ul>
+
+      <div class="card-footer text-center border-top-0 py-2">
+        <span class="form-check d-inline">
+          <input type="checkbox" class="form-check-input" id="certify-pledge" />
+          <label class="form-check-label font-weight-bold" for="certify-pledge">
+            I certify and pledge the above.
+          </label>
+        </span>
+      </div>
+    </div>
+  `;
+}
+
+function StudentGroupControls({ resLocals }: { resLocals: Record<string, any> }) {
+  const { groupSize, groupConfig, startAllowed, user, __csrf_token, notInGroup, assessment } =
+    resLocals;
+  if (notInGroup) {
+    return GroupCreationJoinForm({ groupConfig, __csrf_token });
+  }
+
+  return html`
+    ${renderEjs(
+      import.meta.url,
+      "<%- include('../partials/groupWorkInfoContainer.ejs'); %>",
+      resLocals,
+    )}
```

Comment:
This partial is still in use in the student assessment instance page. Once #9974 is merged, this (and groupRoleTable, and groupRoleErrors) can be converted as well, likely in a future PR.

Reply:
That PR has now been merged! Agreed that we can save that for a future PR. Hopefully your script will surface those as candidates?

Reply:
It already had when I tested with both merged locally.